filters/ExpressionFilter.cpp

```diff
@@ -82,48 +82,70 @@ void ExpressionFilter::addArgs(ProgramArgs& args)
 {
     m_args->m_whereArg = &args.add("expression",
         "Conditional expression describing points to be passed to this filter",
-        m_args->m_expression).setPositional();
+        m_args->m_expressions).setPositional();
 }
 
 
 void ExpressionFilter::prepared(PointTableRef table)
 {
-     if (!m_args->m_expression.valid())
-     {
-         std::stringstream oss;
-         oss << "The expression '" <<  m_args->m_expression
-             << "' is invalid";
-         throwError(oss.str());
-     }
-
-     auto status = m_args->m_expression.prepare(table.layout());
+    for (auto& expression: m_args->m_expressions)
+    {
+        if (!expression.valid())
+        {
+            std::stringstream oss;
+            oss << "The expression '" <<  expression
+                << "' is invalid";
+            throwError(oss.str());
+        }
+
+        auto status = expression.prepare(table.layout());
         if (!status)
             throwError("Invalid 'where': " + status.what());
+    }
 }
 
 
 bool ExpressionFilter::processOne(PointRef& point)
 {
-    bool status = m_args->m_expression.eval(point);
+    if (!(m_args->m_expressions.size() == 1))
+        throwError("Streaming of expressions only works with a single expression");
+
+    bool status = m_args->m_expressions[0].eval(point);
     return status;
 }
 
 
 PointViewSet ExpressionFilter::run(PointViewPtr inView)
 {
     PointViewSet viewSet;
-    PointViewPtr outView = inView->makeNew();
-
     if (!inView->size())
         return viewSet;
 
-    for (PointRef p : *inView)
+    // make a view for each one of our expressions
+    std::map<expr::ConditionalExpression*, PointViewPtr> views;
+    for (expr::ConditionalExpression& expr: m_args->m_expressions)
     {
-        if (processOne(p))
-            outView->appendPoint(*inView, p.pointId());
+        PointViewPtr outView = inView->makeNew();
+        views.insert(std::make_pair(&expr, outView));
     }
 
-    viewSet.insert(outView);
+    // eval our expression across each point for each view.
+    // TODO: make this threaded
+    for(auto& p: views)
```

Comment:
Had a comment here about reversing order, but I'm not sure where it went. You don't want to loop points more than once and you want to avoid using a map if possible. Performance reasons.

Reply:
addressed